src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1090,6 +1090,12 @@ def _get_default_address_pool(cli_ctx, resource_group, balancer_name, balancer_t
     return values[0]
 
 
+def _validate_vmss_single_placement_group(namespace):
+    if namespace.zones or namespace.instance_count > 100:
+        if namespace.single_placement_group is None:
+            namespace.single_placement_group = False
```

Comment:
add comments to call out this is a workaround, considering service team is fixing to default to `false`; you might consider to open a tracking issue